appeals/web/src/server/appeals/appeal-details/withdrawal/withdrawal.controller.js

```diff
@@ -0,0 +1,369 @@
+import { postWithdrawalRequest } from './withdrawal.service.js';
+import logger from '#lib/logger.js';
+import { objectContainsAllKeys } from '#lib/object-utilities.js';
+import { STAGE, DOCTYPE } from '@pins/appeals/constants/documents.js';
+import {
+	postDocumentUpload,
+	postUploadDocumentsCheckAndConfirm,
+	renderDocumentUpload
+} from '../../appeal-documents/appeal-documents.controller.js';
+import { addDocumentDetailsFormDataToFileUploadInfo } from '../../appeal-documents/appeal-documents.mapper.js';
+import { getDocumentRedactionStatuses } from '../../appeal-documents/appeal.documents.service.js';
+import {
+	checkAndConfirmPage,
+	dateWithdrawalRequestPage,
+	manageWithdrawalRequestFolderPage,
+	withdrawalConfirmationPage,
+	withdrawalDocumentRedactionStatusPage
+} from './withdrawal.mapper.js';
+
+/** @type {import('@pins/express').RequestHandler<Response>}  */
+export const getViewWithdrawalDocumentFolder = async (request, response) => {
+	const { errors, currentAppeal } = request;
+
+	if (!currentAppeal || currentAppeal.appealStatus !== 'withdrawn') {
+		return response.status(404).render('app/404');
+	}
+
+	request.currentFolder = currentAppeal.withdrawal?.withdrawalFolder;
+
+	const mappedPageContent = manageWithdrawalRequestFolderPage(
+		`/appeals-service/appeal-details/${request.params.appealId}`,
+		currentAppeal.withdrawal?.withdrawalFolder,
+		currentAppeal.withdrawal?.withdrawalRequestDate,
+		request,
+		'Appeal withdrawal request'
+	);
+
+	return response.status(200).render('appeals/documents/manage-folder.njk', {
+		pageContent: mappedPageContent,
+		errors
+	});
+};
+
+/**
+ * @param {import('@pins/express/types/express.js').Request} request
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
+ */
+export const getWithdrawalRequestUpload = async (request, response) => {
+	return renderWithdrawalRequestUpload(request, response);
+};
+
+/** @type {import('@pins/express').RequestHandler<Response>} */
+export const postWithdrawalRequestRequestUpload = async (request, response) => {
+	const { currentAppeal } = request;
+
+	if (!currentAppeal || currentAppeal.appealStatus === 'withdrawn') {
+		return response.status(404).render('app/404');
+	}
+
+	request.currentFolder = {
+		folderId: currentAppeal.withdrawal?.withdrawalFolder?.folderId,
+		path: `${STAGE.APPELLANT_CASE}/${DOCTYPE.APPELLANT_CASE_WITHDRAWAL}`
+	};
+
+	postDocumentUpload(
```

Comment:
Should this be awaited? and `renderDocumentUpload` below

Reply:
Interesting point. I know we had a recent document refactor throughout the whole platform and we are not awaiting it anywhere else. I'll open conversation about this

Reply:
As agreed, this won't cause an issue in these particular cases, but could be confusing going forward and isn't great practice. We'll `await` these calls in this PR, and create a ticket for any existing code.

Reply:
Done